eng/pipelines/templates/jobs/archetype-sdk-nightly.yml

```diff
@@ -88,38 +72,52 @@ jobs:
         ToxTestEnv: "whl,sdist"  
     - template: ../steps/publish-coverage.yml
 
-  # Run PyPy tests without coverage
-  - job: 'Test_PyPy'
-
+  - job: 'Test_No_Coverage'
     variables:
     - template: ../variables/globals.yml
-    - name: OSName
-      value: 'Linux'
-    - name: PythonVersion
-      value: 'pypy3'
-    - name: OSVmImage
-      value: ubuntu-16.04
 
     dependsOn:
-      - 'Build'
+       - 'Build'
 
     timeoutInMinutes: 240
-    
+
+    strategy:
+      matrix:
+        Linux_Python36:
+          OSName: 'Linux'
+          OSVmImage: 'ubuntu-16.04'
+          PythonVersion: '3.6'
+        Linux_Python35:
+          OSName: 'Linux'
+          OSVmImage: 'ubuntu-16.04'
+          PythonVersion: '3.5'
+        Linux_PyPy3:
+          OSName: 'Linux'
+          OSVmImage: 'ubuntu-16.04'
+          PythonVersion: 'pypy3'
+        MacOS_Python35:
```

Comment:
These names should match the OS and Python versions

Probably should be 
`Windows_Python27` and `MacOS_Python35` respectively

Reply:
Yeah this is fixed in another commit I haven't pushed yet. Thanks for the catch.

Reply:
(Attempting to get rid of duplication, only to hit template insertion errors)